pkg/operation/botanist/health_check.go

```diff
@@ -306,39 +310,52 @@ func (b *HealthChecker) CheckSystemComponents(
 	if err != nil {
 		return nil, err
 	}
-	if exitCondition := b.checkRequiredDeployments(condition, common.RequiredSystemComponentDeployments, deploymentList); exitCondition != nil {
+
+	var (
+		requiredSystemComponentDeployments = utils.Copy(common.RequiredSystemComponentDeployments)
+		requiredSystemComponentDaemonSets  = utils.Copy(common.RequiredSystemComponentDaemonSets)
+	)
+
+	if isKonnectivityTunnelEnabled {
```

Comment:
You need the same Gardener version check here like below - only if the shoot was already reconciled with 1.5 then you may add the `konnectivity-agent` (otherwise, the `vpn-shoot` is still deployed and must be checked).

You could compute whether konnectivity-agent or vpn-shoot must be checked only once in the `checkSystemComponents` function and then pass it here to `CheckSystemComponents` and use it below in `checkSystemComponents` itself.